features/environment.py

```diff
@@ -459,6 +463,37 @@ def before_scenario(context: Context, scenario: Scenario):
     reason = _should_skip_tags(context, scenario.effective_tags)
     if reason:
         scenario.skip(reason=reason)
+        return
+
+    filter_series = context.config.filter_series
+    given_a_series_match = re.match(
+        "a `(.*)` machine with ubuntu-advantage-tools installed",
+        scenario.steps[0].name,
+    )
+    if filter_series and given_a_series_match:
+        series = given_a_series_match.group(1)
+        if series and series not in filter_series:
+            scenario.skip(
+                reason=(
+                    "Skipping scenario outline series `{series}`."
+                    " Cmdline provided @series tags: {cmdline_series}".format(
+                        series=series, cmdline_series=filter_series
+                    )
+                )
+            )
+            return
+
+    if "check.version" in scenario.effective_tags:
+        if context.config.check_version is None:
+            scenario.skip(reason="UACLIENT_BEHAVE_CHECK_VERSION not set")
+            return
+        else:
+            # before_step doesn't execute early enough to modify the step
+            # so we perform the version step surgery here
+            version_step = [s for s in scenario.steps if s.text == "VERSION"][
```

Comment:
Let's add bookend decorators around the VERSION variable name to more clearly indicate that this is a variable intended for substitution  like `%VERSION%` or `{VERSION}`.
Also, can we rename this more explicitly `UACLIENT_BEHAVE_CHECK_VERSION` so we know where that variable is sourced from.  

I can think of other use-cases where we might want to generally replace structured variable names text with something from the getattr(context.config, attr_name) if we come across `{SOME_VAR}` in text. Such as validating properly configured ppa based on UACLIENT_BEHAVE_PPA or UACLIENT_BEHAVE_PPA_KEYID

Reply:
Here's what I was thinking as an example: we'd be able to extend it a bit with a general variable-replacement if needed in the future by adding other "uses.config" decorators.
```diff
diff --git a/features/_version.feature b/features/_version.feature
index 2a32854..95c8a92 100644
--- a/features/_version.feature
+++ b/features/_version.feature
@@ -1,13 +1,13 @@
 Feature: UA is expected version
 
     @series.all
-    @check.version
+    @uses.config.check_version
     Scenario Outline: Check ua version
         Given a `<release>` machine with ubuntu-advantage-tools installed
         When I run `ua version` with sudo
         Then I will see the following on stdout
         """
-        VERSION
+        {UACLIENT_BEHAVE_CHECK_VERSION}
         """
         Examples: version
             | release |
diff --git a/features/environment.py b/features/environment.py
index 21c606a..cf12cb1 100644
--- a/features/environment.py
+++ b/features/environment.py
@@ -483,17 +483,17 @@ def before_scenario(context: Context, scenario: Scenario):
             )
             return
 
-    if "check.version" in scenario.effective_tags:
-        if context.config.check_version is None:
-            scenario.skip(reason="UACLIENT_BEHAVE_CHECK_VERSION not set")
-            return
-        else:
-            # before_step doesn't execute early enough to modify the step
-            # so we perform the version step surgery here
-            version_step = [s for s in scenario.steps if s.text == "VERSION"][
-                0
-            ]
-            version_step.text = context.config.check_version
+    if "uses.config.check_version" in scenario.effective_tags:
+        # before_step doesn't execute early enough to modify the step
+        # so we perform the version step surgery here
+        for s in scenario.steps:
+            if not getattr(s, 'text'):
+                continue
+            if "{UACLIENT_BEHAVE_CHECK_VERSION}" in s.text:
+                s.text = s.text.replace(
+                    "{UACLIENT_BEHAVE_CHECK_VERSION}",
+                    context.config.check_version,
+                )
 
 
 FAILURE_FILES = (
```

Reply:
Really good points! Yep I'll make these suggested changes.

I also need to make sure the version test has all the tags necessary to be run for each test suite